File: frontend/next-terminal/src/features/recommendations/api/recommendations.ts
import { apiGet } from "@/lib/api-client";

export type RecommendationPayload = { recommendation: Record<string, unknown> };

export function fetchLatestRecommendation(symbol: string, timeframe: string) {
  return apiGet<RecommendationPayload>(`/recommendations/latest?symbol=${symbol}&timeframe=${timeframe}`);
}

File: api/routers/recommendations.py
"""Recommendation endpoints."""

from __future__ import annotations

from fastapi import APIRouter, Depends

from api.dependencies import get_dashboard_service
from api.schemas import RecommendationEnvelope
from ui.dashboard_service import DashboardService

router = APIRouter(prefix="/recommendations", tags=["recommendations"])


@router.get("/latest", response_model=RecommendationEnvelope)
def latest(symbol: str = "EURUSD", timeframe: str = "M5", service: DashboardService = Depends(get_dashboard_service)) -> RecommendationEnvelope:
    recommendation = service.generate_recommendation(symbol=symbol, timeframe=timeframe)
    return RecommendationEnvelope(recommendation=service.recommendation_to_record(recommendation))
